File: app/src/main/java/com/mediaproj/momo/ui/main/TutorialActivity.java
package com.mediaproj.momo.ui.main;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.mediaproj.momo.R;
import com.mediaproj.momo.ui.login.LoginActivity;

import me.relex.circleindicator.CircleIndicator;

public class TutorialActivity extends AppCompatActivity {

    FragmentPagerAdapter adapterViewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tutorial);

        //슬라이더
        ViewPager vpPager = (ViewPager) findViewById(R.id.vpPager);
        adapterViewPager = new MyPagerAdapter(getSupportFragmentManager());
        vpPager.setAdapter(adapterViewPager);

        CircleIndicator indicator = (CircleIndicator) findViewById(R.id.indicator);
        indicator.setViewPager(vpPager);

        Button startButton = findViewById(R.id.mainStartButton);
        startButton.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view){
                Intent loginIntent = new Intent(TutorialActivity.this, LoginActivity.class);
                TutorialActivity.this.startActivity(loginIntent);
            }
        });

    }

    public static class MyPagerAdapter extends FragmentPagerAdapter {
        private static int NUM_ITEMS = 4;

        public MyPagerAdapter(FragmentManager fragmentManager) {
            super(fragmentManager);
        }

        // Returns total number of pages
        @Override
        public int getCount() {
            return NUM_ITEMS;
        }

        // Returns the fragment to display for that page
        @Override
        public Fragment getItem(int position) {
            switch (position) {
                case 0:
                    return FirstFragment.newInstance(0, "Page # 1");
                case 1:
                    return SecondFragment.newInstance(1, "Page # 2");
                case 2:
                    return ThirdFragment.newInstance(2, "Page # 3");
                case 3:
                    return fourthFragment.newInstance(3, "Page # 4");
                default:
                    return null;
            }
        }

        // Returns the page title for the top indicator
        @Override
        public CharSequence getPageTitle(int position) {
            return "Page " + position;
        }

    }


}

File: app/src/main/java/com/mediaproj/momo/ui/main/TutorialActivity.kt
//package com.mediaproj.momo.ui.main
//
//import android.content.Intent
//import android.os.Bundle
//import android.widget.Button
//import androidx.appcompat.app.AppCompatActivity
//import androidx.fragment.app.FragmentPagerAdapter
//import me.relex.circleindicator.CircleIndicator
//
//class TutorialActivity: AppCompatActivity() {
//    var adapterViewPager: FragmentPagerAdapter? = null
//    protected fun onCreate(savedInstanceState: Bundle?) {
//        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)
//        //슬라이더
//        val vpPager: ViewPager = findViewById(R.id.vpPager) as ViewPager
//        adapterViewPager = MainActivity.MyPagerAdapter(getSupportFragmentManager())
//        vpPager.setAdapter(adapterViewPager)
//        val indicator = findViewById(R.id.indicator) as CircleIndicator
//        indicator.setViewPager(vpPager)
//        val startButton: Button = findViewById(R.id.mainStartButton)
//        startButton.setOnClickListener {
//            val loginIntent = Intent(this@MainActivity, LoginActivity::class.java)
//            this@MainActivity.startActivity(loginIntent)
//        }
//    }
//
//    class MyPagerAdapter(fragmentManager: FragmentManager?) :
//        FragmentPagerAdapter(fragmentManager) {
//        // Returns total number of pages
//        val count: Int
//            get() = MainActivity.MyPagerAdapter.Companion.NUM_ITEMS
//
//        // Returns the fragment to display for that page
//        fun getItem(position: Int): Fragment? {
//            return when (position) {
//                0 -> FirstFragment.newInstance(0, "Page # 1")
//                1 -> SecondFragment.newInstance(1, "Page # 2")
//                2 -> ThirdFragment.newInstance(2, "Page # 3")
//                3 -> fourthFragment.newInstance(3, "Page # 4")
//                else -> null
//            }
//        }
//
//        // Returns the page title for the top indicator
//        fun getPageTitle(position: Int): CharSequence {
//            return "Page $position"
//        }
//
//        companion object {
//            private const val NUM_ITEMS = 4
//        }
//    }
//}
//4